controller/list_controller.php

```diff
@@ -29,85 +29,111 @@ public function ideas_list($sort)
 			throw new http_exception(404, 'IDEAS_NOT_AVAILABLE');
 		}
 
-		// Build the breadcrumb off the $sort parameter
-		$breadcrumb = (in_array($sort, array(ideas::SORT_NEW, ideas::SORT_TOP, ideas::SORT_IMPLEMENTED)) ? $sort : array());
+		// Overwrite the $sort parameter if the url contains a sort query.
+		// This is needed with the sort by options form at the footer of the list.
+		$sort = ($this->request->is_set('sort')) ? $this->request->variable('sort', ideas::SORT_NEW) : $sort;
 
+		// Get additional query values the url may contain
+		$sort_direction = $this->request->variable('sd', 'd');
+		$status = $this->request->variable('status', 0);
+		$start = $this->request->variable('start', 0);
+
+		// Store original query params for use in breadcrumbs & pagination
+		$u_sort = $sort;
```

Comment:
Yes, just because there are a few places below where$sort gets changed. It's simpler just to store all the original alues here, whether or not they do get changed based on the conditions. This way we know the $u_ variants are always the original input.